backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/search/ScrolledSearch.java

```diff
@@ -20,68 +20,61 @@
 package org.apache.james.backends.es.search;
 
 import java.io.Closeable;
-import java.io.IOException;
 import java.util.Arrays;
 import java.util.Iterator;
-import java.util.concurrent.CompletableFuture;
 import java.util.stream.Stream;
 
-import org.apache.james.backends.es.ListenerToFuture;
+import org.apache.james.backends.es.ReactorElasticSearchClient;
 import org.apache.james.util.streams.Iterators;
 import org.elasticsearch.action.search.ClearScrollRequest;
 import org.elasticsearch.action.search.SearchRequest;
 import org.elasticsearch.action.search.SearchResponse;
 import org.elasticsearch.action.search.SearchScrollRequest;
 import org.elasticsearch.client.RequestOptions;
-import org.elasticsearch.client.RestHighLevelClient;
 import org.elasticsearch.common.unit.TimeValue;
 import org.elasticsearch.search.SearchHit;
 
-import com.github.fge.lambdas.Throwing;
+import reactor.core.publisher.Mono;
 
 public class ScrolledSearch {
```

Comment:
Can't this be represented with a Flux<ScrolledSearch> generated with keeping a state (the scrollId?)

Reply:
I have a branch locally with an intent to do that, I will share it tomorrow (but some tests are failing)